libs/platform/ebpf_platform.h

```diff
@@ -258,11 +258,9 @@ extern "C"
 
     /**
      * @brief Query the platform for the total number of CPUs.
-     * @param[out] cpu_count Pointer to memory location that contains the
-     *    number of CPUs.
+     * @return The count of logical cores in the system.
      */
-    void
-    ebpf_get_cpu_count(_Out_ uint32_t* cpu_count);
+    _Ret_range_(>, 0) uint32_t ebpf_get_cpu_count();
```

Comment:
Also add same annotation to line 278

Reply:
I am not quite sure how to annotate ebpf_get_current_cpu return value as being in range [0, ebpf_get_cpu_count()).

It looks like the values need to be known at compile time.

Reply:
just say > 0

Reply:
Never mind, I misunderstood, 0 is legal.